sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2807,7 +2829,7 @@ public void verifyDeterministic() throws NonDeterministicException {
         SerializableFunction<BoundedWindow, TableReference> tableRefFunction) {
       checkArgument(table == null ^ tableRefFunction == null,
           "Exactly one of table or tableRefFunction should be set");
-      if (table != null) {
+      if (table != null && table.get() != null) {
```

Comment:
not sure this is right. `table.get()` will throw without the check on the next line that `table.isAccessible()`. What is the intended change?